Typos 3.7 (#49)
mulesoft/docs-connector-devkit: modules/ROOT/pages/creating-a-soap-connector.adoc

}
----

==== Multiple WSDLProviders Restriction

When declaring multiple WSDLProvider strategies, all must be consistent in the ServiceDefinitionRetriever declaration.

This includes:

* Return type must be the same for all the strategies, that is either all return List<ServiceDefinition> or all return ServiceDefinition.
* If overridden, labels must be the same for all the retrievers.

> * If overridden, keySeparator must be the same for all the retrievers.

Reviewer: ```suggestion
* If overridden, the keySeparator must be the same for all the retrievers.
```

}
----

> The key part about this strategies, within @WsdlSecurity, is that a connector developer must rely on the @Configurables already in place. This means that if it want to use username token profile, then it must have a way to parametrize the connector with username, password, etc. Once the concrete instances of @WsdlSecurityStrategy initialize, DevKit takes care of the rest, parameterizing each to the underlying engine in the Web Service Consumer.

Reviewer: ```suggestion
The key part about this strategies, within @WsdlSecurity, is that a connector developer must rely on the @Configurables already in place. To use the username token profile, there must be a way to parametrize the connector with a username, a password, and so on. Once the concrete instances of @WsdlSecurityStrategy initialize, DevKit takes care of the rest, parameterizing each to the underlying engine in the Web Service Consumer.
```
